src/transformers/models/encoder_decoder/configuration_encoder_decoder.py

```diff
@@ -74,9 +74,11 @@ class EncoderDecoderConfig(PretrainedConfig):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        assert (
-            "encoder" in kwargs and "decoder" in kwargs
-        ), "Config has to be initialized with encoder and decoder config"
+        if "encoder" not in kwargs or "decoder" not in kwargs:
+            raise ValueError(
```

Comment:
I added a try/except for composite models, catching `ValueError`, which led to to this discovery :) You'll see this pattern fixed in other composite models.